File: trivia.pl
#!/usr/bin/perl

# Trivia game in Perl

# Define the trivia questions and answers
my %trivia = (
    "What is the capital of France?" => "Paris",
    "What is the largest mammal in the world?" => "Blue whale",
    "What is the smallest country in the world?" => "Vatican City",
    "What is the highest mountain in the world?" => "Mount Everest",
    "What is the largest country in the world by area?" => "Russia",
);

# Shuffle the questions randomly
my @questions = keys %trivia;
@questions = sort { rand() <=> 0.5 } @questions;

# Keep track of the user's score
my $score = 0;

# Loop through the questions and ask them one by one
foreach my $question (@questions) {
    # Print the question and ask for the user's answer
    print "$question\n";
    my $answer = <STDIN>;
    chomp($answer);

    # Check the user's answer and update the score accordingly
    if ($answer eq $trivia{$question}) {
        print "Correct!\n";
        $score++;
    } else {
        print "Incorrect. The answer is $trivia{$question}.\n";
    }
}

# Print the final score
print "Your final score is $score out of " . scalar(@questions) . ".\n";

File: trivia.r
# Trivia game in R

# Define questions and answers
questions <- c("What is the capital of Canada?",
               "What is the largest organ in the human body?",
               "What year did World War II end?",
               "Who painted the Mona Lisa?",
               "What is the smallest planet in our solar system?",
               "What is the highest mountain in Africa?",
               "What is the name of the first man to walk on the moon?",
               "What is the chemical symbol for gold?",
               "What is the most abundant gas in the Earth's atmosphere?",
               "What is the largest continent by land area?")

answers <- c("Ottawa", "Skin", "1945", "Leonardo da Vinci", "Mercury", "Mount Kilimanjaro", "Neil Armstrong", "Au", "Nitrogen", "Asia")

# Shuffle questions
questions <- sample(questions)

# Initialize score
score <- 0

# Loop through questions
for (i in 1:length(questions)) {
  
  # Display question
  cat("\nQuestion", i, ":", questions[i], "\n")
  
  # Get user input
  user_answer <- readline(prompt = "Your answer: ")
  
  # Check if answer is correct
  if (tolower(user_answer) == tolower(answers[i])) {
    score <- score + 1
    cat("Correct! Your score is now", score, "\n")
  } else {
    cat("Incorrect. The correct answer is", answers[i], "\n")
  }
}

# Display final score
cat("\nYour final score is", score, "out of", length(questions), "\n")
